dask/utils.py

```diff
@@ -162,6 +162,7 @@ def import_required(mod_name, error_msg):
         raise RuntimeError(error_msg) from e
 
 
+@_deprecated(use_instead="tmp_path or tmp_path_factory from pytest fixtures")
```

Comment:
Since `tmpfile` might be used outside of tests, something like

```python
use_instead="the tempfile module from the standard library or pytest fixtures for temporary files"
```

will cover more potential use cases